aws-android-sdk-s3/src/main/java/com/amazonaws/mobileconnectors/s3/transferutility/TransferUtility.java

```diff
@@ -597,6 +603,48 @@ public TransferObserver upload(String key, File file, ObjectMetadata metadata, C
         return upload(getDefaultBucketOrThrow(), key, file, metadata, cannedAcl, listener);
     }
 
+    /**
+     * Starts uploading the inputStream to the <b>default</b> bucket, using the given key.
+     *
+     * @param key           The key in the specified bucket by which to store the new object.
+     * @param inputStream   The input stream to upload.
+     * @return A TransferObserver used to track upload progress and state
+     */
+    public TransferObserver upload(String key, InputStream inputStream) throws IOException {
+        return upload(key, inputStream, UploadOptions.builder().build());
+    }
+
+    /**
+     * Starts uploading the inputStream to the given bucket, using the given key.
+     *
+     * @param key           The key in the specified bucket by which to store the new object.
+     * @param inputStream   The input stream to upload.
+     * @param options       An UploadOptions which hold all of the optional parameters
+     *                      i.e. bucket, metadata, cannedAcl and transferListener.
+     * @return A TransferObserver used to track upload progress and state
+     */
+    public TransferObserver upload(String key, InputStream inputStream, UploadOptions options) throws IOException {
+        File file = writeInputStreamToFile(inputStream);
+        TransferObserver observer = upload(
+                options.getBucket() != null ? options.getBucket() : getDefaultBucketOrThrow(),
+                key,
+                file,
+                options.getMetadata() != null ? options.getMetadata() : new ObjectMetadata(), options.getCannedAcl(),
+                options.getTransferListener()
+        );
+        deleteTempFile(observer);
```

Comment:
We can't delete the `File` until the upload has completed, since the `File` is read during the upload process.  I would suggest putting this code inside the `removeTransferRecordFromDB` method.